### ISSUE TITLE
no impl PartialEq<Owned> for Box<Ref>

### ISSUE THREAD

**Issue by Emilgardis** [neoeinstein/aliri_braid#22]
`aliri_braid` 0.2.4 doesn't seem to provide this implementation anymore

**Comment by neoeinstein** [comment]
This should have been removed as part of a major version change, so sorry for the unidentified breakage. This was done in order to reduce the number of excess trait impls that didn't have an analogue to `String`/`str`. As there is no `impl PartialEq<String> for Box<str>`, I removed it from being produced in braids.

The workaround for this in the braid and the `String`/`Box<str>` case is to reborrow the value in the box:

```rust
let boxed = String::from("boxed").into_boxed_str();
let owned = String::from("owned");

assert!(&*boxed != owned);
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=393936286d21f76ea4ea29b212f4be17)
